### PR TITLE
Newton error propagation based convergence check.

### PR DESCRIPTION
**Pull request by 1uc**
Since we have access to the Jacobian, we can use Newton error propagation:
```
err = sum(J**2 * sd**2) ** 0.5
```
with `sd = eps * X` to estimate the impact of roundoff errors on `F`.

### CODE REVIEW

File: src/solver/newton/newton.hpp [hunk 1]
```diff
@@ -34,8 +34,22 @@ namespace newton {
  * @{
  */
 
-static constexpr int MAX_ITER = 1e3;
-static constexpr double EPS = 1e-12;
+static constexpr int MAX_ITER = 50;
+static constexpr double EPS = 1e-13;
```
Comment (matz-e) [inline src/solver/newton/newton.hpp:38]: Why the change here?

File: src/solver/newton/newton.hpp [hunk 2]
```diff
@@ -110,7 +121,7 @@ EIGEN_DEVICE_FUNC int newton_solver_small_N(Eigen::Matrix<double, N, 1>& X,
     while (++iter < max_iter) {
         functor(X, F, J);
         double error = F.norm();
```
Comment (JCGoran) [inline src/solver/newton/newton.hpp:123]:
```suggestion
```

Think we don't need this one anymore.

**Review by matz-e — APPROVED** [review]
Looks sensible.

File: src/solver/newton/newton.hpp (hunk 1 above)
Comment (1uc) [reply src/solver/newton/newton.hpp:38]: The one for MAX_ITER is a compromise between getting closer to NRN and allowing infinite guessing. If Newton converges it converges extremely fast, in every iteration the number of `0` doubles. The problem tends to be that it doesn't converge. However, if it doesn't more iterations are unlikely to help. Either one's stuck just above EPS, or it's paying ping pong, or it's diverging.

The EPS is needed to make the tests pass the check that the residual is under a certain threshold.

Comment (matz-e) [reply src/solver/newton/newton.hpp:38]: Was just wondering why you lowered the EPS. Thanks for elaborating on the much lower MAX_ITER.

File: src/solver/newton/newton.hpp (hunk 2 above)
Comment (1uc) [reply src/solver/newton/newton.hpp]: Thank you, gone.
